app/src/page/channel/view.js

```diff
@@ -1,20 +1,52 @@
 // @flow
 import React from 'react';
-import { ScrollView, Text, View } from 'react-native';
+import { ActivityIndicator, Text, View } from 'react-native';
+import Colors from '../../styles/colors';
+import FileList from '../../component/fileList';
+import PageHeader from '../../component/pageHeader';
 import UriBar from '../../component/uriBar';
 import channelPageStyle from '../../styles/channelPage';
 
 class ChannelPage extends React.PureComponent {
+  componentDidMount() {
+    const { uri, page, fetchClaims, fetchClaimCount } = this.props;
+
+    fetchClaims(uri, page || 1);
```

Comment:
You could check if you already have the claims and only fetch if they don't exist.